Sources/ProjectSpec/Breakpoint.swift

```diff
@@ -0,0 +1,162 @@
+
+    public init(extensionID: String,
+                enabled: Bool = true,
+                ignoreCount: String = "0",
+                continueAfterRunningActions: Bool = false,
+                filePath: String? = nil,
+                timestamp: String? = nil,
+                startingColumn: String? = nil,
+                endingColumn: String? = nil,
+                startingLine: String? = nil,
+                endingLine: String? = nil,
+                breakpointStackSelectionBehavior: String? = nil,
+                symbol: String? = nil,
+                module: String? = nil,
+                scope: String? = nil,
+                stopOnStyle: String? = nil,
+                actions: [Breakpoint.Action] = [],
+                locations: [Breakpoint.Location] = []) {
+        self.extensionID = extensionID
+        self.enabled = enabled
+        self.ignoreCount = ignoreCount
+        self.continueAfterRunningActions = continueAfterRunningActions
+        self.filePath = filePath
+        self.timestamp = timestamp
+        self.startingColumn = startingColumn
+        self.endingColumn = endingColumn
+        self.startingLine = startingLine
+        self.endingLine = endingLine
+        self.breakpointStackSelectionBehavior = breakpointStackSelectionBehavior
+        self.symbol = symbol
+        self.module = module
+        self.scope = scope
+        self.stopOnStyle = stopOnStyle
+        self.actions = actions
+        self.locations = locations
+    }
+
+    public static func == (lhs: Breakpoint, rhs: Breakpoint) -> Bool {
+        return lhs.extensionID == rhs.extensionID &&
+        lhs.enabled == rhs.enabled &&
+        lhs.ignoreCount == rhs.ignoreCount &&
+        lhs.continueAfterRunningActions == rhs.continueAfterRunningActions
+    }
+}
+
+extension Breakpoint.Action: JSONObjectConvertible {
+    public init(jsonDictionary: JSONDictionary) throws {
+        if let extensionID: String = jsonDictionary.json(atKeyPath: "extensionID") {
+            self.extensionID = extensionID
+        } else {
+            throw SpecParsingError.invalidBreakpoint(jsonDictionary)
+        }
+        consoleCommand = jsonDictionary.json(atKeyPath: "consoleCommand")
+        message = jsonDictionary.json(atKeyPath: "message")
+        conveyanceType = jsonDictionary.json(atKeyPath: "conveyanceType")
+        command = jsonDictionary.json(atKeyPath: "command")
+        arguments = jsonDictionary.json(atKeyPath: "arguments")
+        waitUntilDone = jsonDictionary.json(atKeyPath: "waitUntilDone")
+        script = jsonDictionary.json(atKeyPath: "script")
+        soundName = jsonDictionary.json(atKeyPath: "soundName")
+    }
+}
+
+extension Breakpoint.Location: JSONObjectConvertible {
+    public init(jsonDictionary: JSONDictionary) throws {}
+}
+
+extension Breakpoint: JSONObjectConvertible {
+
+    public init(jsonDictionary: JSONDictionary) throws {
+        if let extensionID: String = jsonDictionary.json(atKeyPath: "extensionID") {
```

Comment:
It's ok to just have extensionID be a required property and just let JSONUtilities throw a decoding error, otherwise we'd have tons of these custom errors

Reply:
Ok, done